aya-log/src/lib.rs

```diff
@@ -108,12 +110,54 @@ impl BpfLogger {
         bpf: &mut Bpf,
         logger: T,
     ) -> Result<BpfLogger, Error> {
-        let logger = Arc::new(logger);
-        let mut logs: AsyncPerfEventArray<_> = bpf
-            .take_map(MAP_NAME)
-            .ok_or(Error::MapNotFound)?
-            .try_into()?;
+        let map = bpf.take_map(MAP_NAME).ok_or(Error::MapNotFound)?;
+        Self::read_logs_async(map, logger)?;
+        Ok(BpfLogger {})
+    }
+
+    /// Attaches to an existing `aya-log-ebpf` log created by a running ebpf
```

Comment:
nit:

```
/// Starts reading log records created by the eBPF program identified by `program_id` and logs them
/// with the default logger. See [log::logger].
///
/// It can be used to read the logs of a pinned program.
```

I wouldn't say anything about background threads.

Reply:
Ok, I've reworked the comment(s) w/o mentioning the thread.